hs_geo_raster_resource/migrations/custom_migration_for_raster_meta_update_20160512.py

```diff
@@ -0,0 +1,139 @@
+from __future__ import unicode_literals
+
+import os
+import shutil
+import logging
+import tempfile
+import subprocess
+
+from django.db import migrations
+from django.core.files.uploadedfile import UploadedFile
+
+from django_irods.storage import IrodsStorage
+from hs_core import hydroshare
+from hs_core.hydroshare.utils import resource_modified
+from hs_geo_raster_resource.models import RasterResource
+from hs_geo_raster_resource import raster_meta_extract
+
+
+def migrate_tif_file(apps, schema_editor):
+    log = logging.getLogger()
+    istorage = IrodsStorage()
+    success = 0
+    fail = []
+    not_modify = []
+
+    for res in RasterResource.objects.all():
+        if res.files.all():
+            # update metadata information
+            try:
+                # copy all the files of a resource to temp dir
+                temp_dir = tempfile.mkdtemp()
+                for res_file in res.files.all():
+                    shutil.copy(res_file.resource_file.file.name,
+                                os.path.join(temp_dir, os.path.basename(res_file.resource_file.name)))
+
+                # update vrt file for single tif file
+                if len(os.listdir(temp_dir)) == 2:
+                    # create new vrt file
+                    tif_file_path = [os.path.join(temp_dir, f) for f in os.listdir(temp_dir) if '.tif' == f[-4:]].pop()
+                    vrt_file_path = [os.path.join(temp_dir, f) for f in os.listdir(temp_dir) if '.vrt' == f[-4:]].pop()
+
+                    try:
+                        with open(os.devnull, 'w') as fp:
+                            subprocess.Popen(['gdal_translate', '-of', 'VRT', tif_file_path, vrt_file_path], stdout=fp, stderr=fp).wait()   # remember to add .wait()
+
+                        # delete vrt res file
+                        for f in res.files.all():
+                            if 'vrt' == f.resource_file.name[-3:]:
+                                f.resource_file.delete()
+                                f.delete()
+
+                        # add new vrt file to resource
+                        new_file = UploadedFile(file=open(vrt_file_path, 'r'), name=os.path.basename(vrt_file_path))
+                        hydroshare.add_resource_files(res.short_id, new_file)
+
+                        # update the bag
+                        bag_name = 'bags/{res_id}.zip'.format(res_id=res.short_id)
+                        if istorage.exists(bag_name):
+                            # delete the resource bag as the old bag is not valid
+                            istorage.delete(bag_name)
+                        resource_modified(res, res.creator)
```

Comment:
@hyi In another custom migration file custom_migration_for_tif_to_vrt_20160223.py , we added these lines. Because after the model migration, the bag was corrupted and the downloaded zip bag file can't be unzipped correctly. These lines helped to solve this issue. So I add them in this migration file for the purpose to make sure the user can download a valid zip file. I will try to remove these lines to test if this issue exists for this custom migration. Thanks